src/metabase/api/dashboard.clj

```diff
@@ -613,9 +671,23 @@
     (dashboard-card/delete-dashboard-cards! dashcard-ids)
     dashboard-cards))
 
+(defn- assert-new-dashcards-are-not-internal-to-other-dashboards [dashboard to-create]
+  (when-let [card-ids (seq (into #{} (concat
```

Comment:
`into #{}` here seems redundant?

Reply:
The idea was to remove duplicates for the `:in` later, but it does seem like `t2` works fine with non-unique collections, e.g. `(count (t2/select :model/Card :id [:in [1 1 2 1 2]]))` returns `2` for me locally. But maybe for clarity I'll just turn this into `(seq (concat ...))` and then do `(set card-ids)` in the query itself.

Reply:
ah I see, yeah that would be clearer

Reply:
agreed